components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/dto/PublisherDataHolder.java

```diff
@@ -99,6 +110,40 @@ public PublisherDataHolder(Resource resource, boolean returnSecrets) {
         this.propertyDTOs = propertyDTOs.toArray(new PublisherPropertyDTO[propertyDTOs.size()]);
     }
 
+    public PublisherDataHolder(ResultSet resultSet, boolean returnSecrets) throws SQLException {
+
+        List<PublisherPropertyDTO> propertyDTOList = new ArrayList<>();
+
+        while (resultSet.next()) {
+            PublisherPropertyDTO dto = new PublisherPropertyDTO();
+
+            dto.setId(resultSet.getString(PROPERTY_ID));
+            dto.setValue(resultSet.getString(PROPERTY_VALUE));
+            dto.setDisplayName(resultSet.getString(DISPLAY_NAME));
+            dto.setDisplayOrder(resultSet.getInt(DISPLAY_ORDER));
+            dto.setRequired(resultSet.getBoolean(IS_REQUIRED));
+            dto.setSecret(resultSet.getBoolean(IS_SECRET));
+            dto.setModule(resultSet.getString(MODULE));
+
+            if (dto.isSecret() && (returnSecrets)) {
+                String password = dto.getValue();
+                try {
+                    password = new String(CryptoUtil.getDefaultCryptoUtil().
+                            base64DecodeAndDecrypt(dto.getValue()));
+                } catch (CryptoException e) {
+                    log.error(e);
+                    // ignore
+                }
+                dto.setValue(password);
+
+            }
+
+            this.moduleName = resultSet.getString(ENTITLEMENT_MODULE_NAME);
```

Comment:
are we getting this moduleName somewhere else? whats the need to setting it here?